src/AimInputComponent.cpp

```diff
@@ -7,48 +7,101 @@
 AimInputComponent::AimInputComponent() {
     toggleXRotation = false;
     toggleYRotation = false;
-    spaceHoldDown = false;
+    toggleThrow = false;
+    triggerAvailable = true;
+    actionHoldDown = false;
 }
 
 AimInputComponent::~AimInputComponent() {
 
 }
 
-// rotate direction with rotate if pressed
-
 void AimInputComponent::pollInput() {
+   if (glfwJoystickPresent(GLFW_JOYSTICK_1)) {
+      pollGamepad();
+   } else {
+      pollKeyboard();
+   }
+}
+
+void AimInputComponent::pollGamepad() {
+   int count;
+   const float* axes = glfwGetJoystickAxes(GLFW_JOYSTICK_1, &count);
+   float xComponent = -axes[3], yComponent = -axes[2];
+   // A trigger at rest has a value of -1.0, fully engaged is 1.0.
+   float leftTrigger = axes[4], rightTrigger = axes[5];
+
+   if (-0.1f < xComponent && xComponent < 0.1f) {
```

Comment:
Hahaha, Reed loving the dead space code XD. Yea I thought it was pretty cool myself haha, weird to see it!